httpd/handler.go

```diff
@@ -175,6 +179,127 @@ func (h *Handler) serveQuery(w http.ResponseWriter, r *http.Request, user *influ
 	httpResults(w, results, pretty)
 }
 
+type Point struct {
+	Name      string                 `json:"name"`
+	Timestamp time.Time              `json:"timestamp"`
+	Tags      map[string]string      `json:"tags"`
+	Fields    map[string]interface{} `json:"fields"`
+}
+
+type Batch struct {
+	Database        string  `json:"database"`
+	RetentionPolicy string  `json:"retentionPolicy"`
+	Points          []Point `json:"points"`
+}
+
+func interfaceToString(v interface{}) string {
+	switch t := v.(type) {
+	case nil:
+		return ""
+	case bool:
+		return fmt.Sprintf("%v", v)
+	case int, int8, int16, int32, int64, uint, uint8, uint16, uint32, uint64, uintptr:
+		return fmt.Sprintf("%d", t)
+	case float32, float64:
+		return fmt.Sprintf("%v", t)
+	default:
+		return fmt.Sprintf("%v", t)
+	}
+}
+
+// Return all the measurements from the given DB
+func (h *Handler) showMeasurements(db string, user *influxdb.User) (error, []string) {
+	var measurements []string
+	queryString := "show measurements"
+	p := influxql.NewParser(strings.NewReader(queryString))
+	query, err := p.ParseQuery()
+	if err != nil {
```

Comment:
For me it was originally about being more performant and less about readability, but after seeing it the readability is probably more important.  Our parser is pretty performant, especially for that type of query.  I originally thought we could pass in that simple type, but having to wrap it that way is pretty ugly, I agree.